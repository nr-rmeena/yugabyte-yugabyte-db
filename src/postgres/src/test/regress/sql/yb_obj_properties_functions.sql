-- Tests for yb_obj_properties_functions: Verify new YB functions to get table
-- or database properties.
--
-- Test NON-COLOCATED database.
--
-- Test yb_is_database_colocated.
SELECT yb_is_database_colocated();

-- Test yb_table_properties.
CREATE TABLE temp_tbl (h1 INT, h2 INT, PRIMARY KEY((h1, h2))) SPLIT INTO 7 TABLETS;

SELECT yb_table_properties('temp_tbl'::regclass);

-- Cleanup.
DROP TABLE temp_tbl;

--
-- Test COLOCATED database.
--
CREATE DATABASE colocated_db colocated = true;

\c colocated_db
--
-- Test yb_is_database_colocated.
SELECT yb_is_database_colocated();
--
-- Test yb_table_properties.
CREATE TABLE col_temp_tbl (h INT PRIMARY KEY) WITH (colocated = true) SPLIT INTO 5 TABLETS;

SELECT yb_table_properties('col_temp_tbl'::regclass);

-- Cleanup.
DROP TABLE col_temp_tbl;
