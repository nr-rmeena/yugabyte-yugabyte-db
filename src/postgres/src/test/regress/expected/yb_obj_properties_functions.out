-- Tests for yb_obj_properties_functions: Verify new YB functions to get table
-- or database properties.
--
-- Test NON-COLOCATED database.
--
-- Test yb_is_database_colocated.
SELECT yb_is_database_colocated();
 yb_is_database_colocated 
--------------------------
 f
(1 row)

-- Test yb_table_properties.
CREATE TABLE temp_tbl (h1 INT, h2 INT, PRIMARY KEY((h1, h2))) SPLIT INTO 7 TABLETS;
SELECT yb_table_properties('temp_tbl'::regclass);
 yb_table_properties 
---------------------
 (7,2,f)
(1 row)

-- Cleanup.
DROP TABLE temp_tbl;
--
-- Test COLOCATED database.
--
CREATE DATABASE colocated_db colocated = true;
\c colocated_db
--
-- Test yb_is_database_colocated.
SELECT yb_is_database_colocated();
 yb_is_database_colocated 
--------------------------
 t
(1 row)

--
-- Test yb_table_properties.
CREATE TABLE col_temp_tbl (h INT PRIMARY KEY) WITH (colocated = true) SPLIT INTO 5 TABLETS;
SELECT yb_table_properties('col_temp_tbl'::regclass);
 yb_table_properties 
---------------------
 (1,0,t)
(1 row)

-- Cleanup.
DROP TABLE col_temp_tbl;
